### DISCUSSION TITLE
SSH keys set up but GitHub still asks for password

### DISCUSSION THREAD

**Discussion by balabin** [community/community#23171] [New to GitHub]
<p>Hi, I have set up and double-checked my SSH keys.; running “ssh -T <a href="mailto:git@github.com">git@github.com</a>” gives a message “You’ve been successfully authenticated …”. However, when trying to run “git push”, I still get the Username/Password prompts. I expected that the SSH keys will function as a password-less authentication mechanism, just like they do on Linux and other UNIXes, but that does not happen. What am I missing? Thanks for your time and help.</p>

**Comment by airtower-luna** [comment]
<p>You’ll have to also adjust the remote URL your local repository uses to access your Github repository. That URL decides which protocol (SSH or HTTPS) Git uses. You can find the URLs behind the green “Code” button on your repository page.</p>
<p>Once you have the one you want, you can use <a href="https://git-scm.com/docs/git-remote"><code>git remote</code></a> to set which URL to use:</p>
<pre><code class="lang-bash">git remote set-url origin &lt;URL&gt;
git remote set-url --push origin &lt;URL&gt;
</code></pre>

**Comment by balabin** [comment]
<p>Thanks a lot. It appears that the first command sets the remote URL as the value for “url” in the [remote “origin”] section of the local .git/config file (which was already set when the local repo was cloned to GitHub), and the second one sets the value for “pushurl” in the same section. However, when running “git push”, I still get asked for the GitHub username and password, despite having SSH keys set up and tested. As such, the question is whether there is a way to set up the local computer so that I can simply “git push” without entering any additional credentials? If not, what is the SOP for “git push” to the GitHub repository then? Thanks for any hint.</p>

**Comment by byrneh** [comment, marked as answer]
<p>Hi <a class="mention" href="/balabin">@balabin</a><br>
A common mistake is using the wrong URL, as this actually determines the authentication mechanism is used. So whether a local configured SSH key is used, or a prompt for username/token or their local credential manager cached values.<br>
You can only push to two types of URL addresses:<br>
An HTTPS URL like <a href="https://github.com/user/repo.git" rel="noopener nofollow ugc">https://github.com/user/repo.git</a><br>
An SSH URL, like <a href="mailto:git@github.com">git@github.com</a>:user/repo.git<br>
I cannot see from your question what URL format you have used, but worth checking.</p>
<p>It not, it sounds like your configuration is wrong, there are excellent step by step instructions in GitHub docs that can be followed, on previous experience working with other users problems following these explicitly has worked for them. If you still experience issues it will help anyone assist you in further problem analysis<br>
Useful Docs links:<br>
<a href="https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh" rel="noopener nofollow ugc">connecting-to-github-with-ssh</a><br>
<a href="https://docs.github.com/en/github/using-git/about-remote-repositories" rel="noopener nofollow ugc">about-remote-repositories</a></p>

**Comment by balabin** [comment]
<p>Hi <a class="mention" href="/byrneh">@byrneh</a>,</p>
<p>Indeed, the issue was having an HTTP URL in .git/config; changing it to the SSH URL made SSH keys work. Thanks for the documentation links, too; somehow, I overlooked the second one. Things are much more usable now - thanks a lot for your help!</p>

**Comment by byrneh** [comment]
<p>Hi <a class="mention" href="/balabin">@balabin</a> glad you are up and working as you want now.<br>
If you are happy it’s answered you can mark the post as solved.</p>

**Comment by balabin** [comment]
<p>Hi <a class="mention" href="/byrneh">@byrneh</a>, done - thanks for the hint.</p>

**Comment by Phobia-Cosmos** [comment]
so I wonder if it's correct that using the HTTPS method to push is banned so we can only use the SSH method now?[@balabin](https://github.com/balabin)

**Reply by airtower-luna** [reply]
You can use HTTPS just fine, just remember you need to use a PAT instead of your password for Git operations over HTTPS. Though personally I'd recommend SSH for *both* security and convenience. :slightly_smiling_face:

**Comment by milliephanillie** [comment]
I have the same issue

I am using the correct ssh url
I have tried using git remote set-url orgin <url>
I have verified the config file has no https
running “ssh -T git@github.com” gives a message “You’ve been successfully authenticated …”.

No matter what I do when i rum remote -v i am seeing the https urls?

My powershell on the windows side is working correctly.

**Reply by airtower-luna** [reply]
Do you maybe have a [`url.<base>.insteadOf`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf) setting for Github in user or global config? See https://git-scm.com/docs/git-config#FILES for possible locations.
